pkg/cnb/project_descriptor.go

```diff
@@ -105,6 +107,17 @@ func processFiles(appDir string, d build) error {
 	})
 }
 
+// Because CNB Project Descriptor v0.2 has two ways for defining environment variables.
+// see https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md#iobuildpacksbuildenv-optional
+// This function calculates the final environment variables
+func getEnvVariables(d descriptorV2) []envVariable {
+	env := d.IO.Buildpacks.BuildEnv.Env
+	if env == nil {
+		env = d.IO.Buildpacks.Env
```

Comment:
let's start the a thread here per: https://github.com/pivotal/kpack/pull/1027#issuecomment-1235771433

> The current 'old' code in kpack seems a bit incorrect? See output section of https://github.com/buildpacks/pack/pull/1479

It appears we incorrectly used ` io.buildpacks.env` while `pack` incorrectly used `io.buildpacks.env.build`.

We can either support all three or ignore support for `io.buildpacks.env.build`

@samj1912 @tomkennedy513 @matthewmcnew @jjbustamante 
for the old key vs the new key

old pack key - io.buildpacks.env.build

new pack key - io.buildpacks.build.env

old kpack key - io.buildpacks.env? (this seems incorrect?)

Reply:
Kpack exists in the pack ecosystem where users would reasonably expect kpack to behave similarly to pack. So, I think we should provide backwards compatibility to both `io.buildpacks.env.build` and kpack incorrect key: `io.buildpacks.env`.

Reply:
Good 👀 @samj1912 !